### PR TITLE
feat(logs): Changes in the engine to enable log events for the outer HTTP requests

### PR DESCRIPTION
**Pull request by jakubadamw**
# Description

This is mostly an internal change that adds another variant to the log event type required for some changes upstream in the platform.

# Type of change

- [ ] 💔 Breaking
- [X] 🚀 Feature
- [ ] 🐛 Fix
- [ ] 🛠️ Tooling
- [ ] 🔨 Refactoring
- [ ] 🧪 Test
- [ ] 📦 Dependency
- [ ] 📖 Requires documentation update

### CODE REVIEW

File: engine/crates/gateway-core/src/lib.rs [hunk 1]
```diff
@@ -45,6 +45,8 @@ pub trait Gateway: Send {
     fn executor(&self) -> &Arc<Self::Executor>;
     fn cache_config(&self) -> Cow<'_, CacheConfig<'_>>;
 
+    fn post_request(&self, _url: &str, _method: http::Method, _status_code: u16, _duration: std::time::Duration) {}
```
[the remark this answers is not in the document]
Comment (Finistere) [reply engine/crates/gateway-core/src/lib.rs:48]: I don't understand why it needs to be part of the Gateway trait to be honest

[the remark this answers is not in the document]
Comment (Finistere) [reply engine/crates/gateway-core/src/lib.rs:48]: the gateway is about executing a request. Logging of request is outside of that scope in my mind at least, and should ideally be done with the help of the http framework typically

[the remark this answers is not in the document]
Comment (jakubadamw) [reply engine/crates/gateway-core/src/lib.rs:48]: Guys, I have no problem with it not being part of the interface. From what I know, we introduced it to get rid of the forest of `#[cfg(feature = "")]` bounds. If we're okay with that, then sure, I'll remove it

[the remark this answers is not in the document]
Comment (jakubadamw) [reply engine/crates/gateway-core/src/lib.rs:48]: Well, actually… I guess I can still do it without a feature `#[cfg()]`, fine

[the remark this answers is not in the document]
Comment (Finistere) [reply engine/crates/gateway-core/src/lib.rs]: the feature flag will be removed with the native gateway work with which I'm almost done with ;)

**Pull request merged**